lib/internal/repl.js

```diff
@@ -127,7 +127,8 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
     }
 
     if (data) {
-      repl.history = data.split(/[\n\r]+/, repl.historySize);
+      repl.history = data.trim().split(/[\n\r]+/)
+        .reverse().slice(0, repl.historySize);
```

Comment:
Does this PR change the ordering in the history file? if so, it is probably `semver-major`.